std/format.d

```diff
@@ -4302,7 +4302,7 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     import std.conv : parse, text;
     if (spec.spec == 's')
     {
-        return parse!T(input);
+        static if(__traits(compiles, parse!T(input)) ) return parse!T(input);
     }
```

Comment:
This puts the runtime test before the compile-time test which is unusual. The more intuitive way would be:

```d
static if (is(typeof(parse!T(input)) == T)) 
{
    if (spec.spec == `s`) return parse!T(input);
}
```

Also note you really want the type returned by `parse` to be `T`, space after `if`, and no space before closed paren.